rules/defense_evasion/disable_apparmor_attempt.yaml

```diff
@@ -10,14 +10,50 @@
+    (payload.filename STARTS_WITH "/usr/bin/systemctl" AND payload.argv CONTAINS "disable" AND (payload.argv CONTAINS "apparmor" OR payload.argv CONTAINS "apparmor.service")) OR
+    (payload.filename ENDS_WITH "service" AND payload.argv CONTAINS "apparmor" AND payload.argv CONTAINS "stop") OR
+    (payload.filename ENDS_WITH "chkconfig" AND payload.argv CONTAINS "apparmor" AND payload.argv CONTAINS "off")
+    )
+    
+- name: Move or symlink AppArmor profile to disable
+  type: Exec
+  category: defense_evasion
+  severity: low
+  description: Monitors for the use of ln or mv to relocate AppArmor profiles into the disable directory, which prevents them from being loaded. 
+    Attackers may attempt to disable AppArmor profiles without fully uninstalling them.
+  condition: (payload.filename STARTS_WITH "/usr/bin/ln" AND payload.argv CONTAINS "/etc/apparmor.d/" AND payload.argv CONTAINS "/etc/apparmor.d/disable/")
+    OR (payload.filename STARTS_WITH "/usr/bin/mv" AND payload.argv CONTAINS "/etc/apparmor.d/" AND payload.argv CONTAINS "/etc/apparmor.d/disable/")
 
-- name: Create symlink to disable AppArmor
+- name: AppArmor configuration file modification
+  type: FileOpened
+  category: defense_evasion
+  severity: low
+  description: Detects attempts to modify AppArmor configuration files which could indicate an attacker trying to disable or weaken AppArmor protections.
+  condition: (
+      payload.filename STARTS_WITH "/etc/apparmor" OR
+      payload.filename STARTS_WITH "/etc/apparmor.d/" OR
+      payload.filename STARTS_WITH "/etc/apparmor.conf"
+    )
+    AND (payload.flags CONTAINS "O_WRONLY" OR payload.flags CONTAINS "O_RDWR") 
+    
+- name: AppArmor kernel parameter modification
+  type: Exec
+  category: defense_evasion
+  severity: low
+  description: Detects attempts to modify kernel parameters related to AppArmor which could disable its protections.
+  condition: payload.filename STARTS_WITH "/usr/sbin/sysctl" AND 
+    (payload.argv CONTAINS "apparmor=0" OR payload.argv CONTAINS "kernel.apparmor.enabled=0" OR payload.argv CONTAINS "security=none")
+    
+- name: AppArmor teardown or profile removal
   type: Exec
   category: defense_evasion
   severity: low
-  description: Monitors for the execution of the ln utility to create a symlink to disable AppArmor. Attackers may attempt to create a 
-    symlink to disable AppArmor to evade detection and avoid possible restrictions on their actions and resources.
-  condition: payload.filename == "/usr/bin/ln" AND payload.argv CONTAINS "/etc/apparmor.d/" AND payload.argv CONTAINS "/etc/apparmor.d/disable/"
\ No newline at end of file
+  description: Detects use of AppArmor utilities to unload profiles or tear down AppArmor protection.
+  condition: (
+    (payload.filename ENDS_WITH "apparmor_parser" AND (payload.argv CONTAINS "-R" OR payload.argv CONTAINS "--remove")) OR
+    (payload.filename ENDS_WITH "aa-teardown") OR
+    (payload.filename ENDS_WITH "aa-disable")
+    )
```

Comment:
check if these utilities utilizes file system under the hood and create a rule on that if it's possible